### DISCUSSION TITLE
Executing same test suite on multiple targets simultaneously

### DISCUSSION THREAD

**Discussion by jeffwright13** [pytest-dev/pytest#10580] [General]
I know about pytest-xdist, which lets me run one test suite on the same target quicker by using multiple CPUs. But what if I wanted to run the same test suite on multiple different targets, all at the same time? Is there a plugin which does this already, or am I looking at rolling my own?

**Comment by RonnyPfannschmidt** [comment]
ts absolutely unclear what you are asking fors even a target?

xdist supports ssh based span out including rsync of the testsuite

**Comment by jeffwright13** [comment]
OK, I will try again.

I have a test suite that contains 100 test cases. I want to run the same test suite against System 1, and System 2, and System 3. To save time, I would like to be able to launch the 3 individual executions of the test suite, against the 3 systems, simultaneously.

The SSH feature of pytest-xdist [is deprecated](https://pytest-xdist.readthedocs.io/en/latest/remote.html). But looking a bit lower on that same page there seems to be a possibility for my use-case: https://pytest-xdist.readthedocs.io/en/latest/remote.html#running-tests-on-many-platforms-at-once

**Comment by jeffwright13** [comment]
Maybe I make this easy on myself and just use GNU Parallel.
